src/main/java/io/dropwizard/sharding/DBShardingBundle.java

```diff
@@ -50,34 +50,53 @@
  * A dropwizard bundle that provides sharding over normal RDBMS.
  */
 public abstract class DBShardingBundle<T extends Configuration> implements ConfiguredBundle<T> {
+
     private List<HibernateBundle<T>> shardBundles = Lists.newArrayList();
     @Getter
     private List<SessionFactory> sessionFactories;
     @Getter
     private ShardManager shardManager;
+    @Getter
+    private String dbNamespace;
+
+    public DBShardingBundle(String dbNamespace, Class<?> entity, Class<?>... entities) {
+        this.dbNamespace = dbNamespace;
 
-    public DBShardingBundle(Class<?> entity, Class<?>... entities) {
         val inEntities = ImmutableList.<Class<?>>builder().add(entity).add(entities).build();
         init(inEntities);
     }
 
-    public DBShardingBundle(String classPathPrefix) {
+    public DBShardingBundle(String dbNamespace, String classPathPrefix) {
+        this.dbNamespace = dbNamespace;
+
         Set<Class<?>> entities = new Reflections(classPathPrefix).getTypesAnnotatedWith(Entity.class);
         Preconditions.checkArgument(!entities.isEmpty(), String.format("No entity class found at %s", classPathPrefix));
         val inEntities = ImmutableList.<Class<?>>builder().addAll(entities).build();
         init(inEntities);
     }
 
+    public DBShardingBundle(Class<?> entity, Class<?>... entities) {
+        this("default", entity, entities);
+    }
+
+    public DBShardingBundle(String classPathPrefix) {
+        this("default", classPathPrefix);
+    }
+
     private void init(final ImmutableList<Class<?>> inEntities) {
-        String numShardsEnv = System.getProperty("db.shards", "2");
+        final String SHARD_ENV = "db.shards";
```

Comment:
Move to class level static final constant.

Reply:
Defined a inner class called shardingKeys. Forgot to push it. :) 

My bad! Pushed it!